### PR TITLE
mention eglot for emacs support

### PR DESCRIPTION
**Pull request by ssnnoo**
eglot is a popular lsp client for emacs that supports fortls.

### CODE REVIEW

**Review by gnikit — CHANGES_REQUESTED** [review]
Thanks for the PR, a few things:
1. Multiple plugins should preferably be in their own subsection see the [Vim section](https://gnikit.github.io/fortls/editor_integration.html#vim) (the Atom section is not a good example since my extension it is a WIP)
2. All the necessary instructions on how to configure `fortls` with eglot should be provided so users can copy-paste them in their configs. It is important to set reasonable options in the config.
3. I don't know how eglot works or if it downloads the server for you but their README link to `fortls` is wrong, it is pointing to the `fortran-language-server` project not this repo.

**Comment by ssnnoo** [comment]
Ok, I will update the pull request tonight. 

Regarding the wrong link please see https://github.com/joaotavora/eglot/pull/965, hopefully this will be updated soon. 

Last but not least: if in f90 mode, eglot will start fortls as default, so no copy and paste required :-)

**Comment by ssnnoo** [comment]
Ok, I updated the pull request. 

Unfortunately, nothing has happend regarding https://github.com/joaotavora/eglot/pull/965 yet.

**Comment by gnikit** [comment]
Any news on this @ssnnoo ?

**Comment by gnikit** [comment]
> > Any news on this @ssnnoo ?
> 
> Hi, I changed the commit as requested a while ago and it should be fine now (hopefully). One thing that did not happen is that [joaotavora/eglot#965](https://github.com/joaotavora/eglot/pull/965) got merged. But this is completely independent of this contribution here :-)

The problem is that the underlying extension is pointing to the wrong server, which leads to confusion from the user's perspective. I will go drop a comment

**Comment by ssnnoo** [comment]
> > > Any news on this @ssnnoo ?
> > 
> > 
> > Hi, I changed the commit as requested a while ago and it should be fine now (hopefully). One thing that did not happen is that [joaotavora/eglot#965](https://github.com/joaotavora/eglot/pull/965) got merged. But this is completely independent of this contribution here :-)
> 
> The problem is that the underlying extension is pointing to the wrong server, which leads to confusion from the user's perspective. I will go drop a comment

Sure, I agree. But this is only in the readme, in the source code it only refers to fortls. 

`((fortran-mode f90-mode) . ("fortls"))` and this should be fine. There is not automatic installation or anything.

File: docs/editor_integration.rst [hunk 1]
```diff
@@ -159,13 +159,26 @@ to set ``fortls`` for Fortran files
...
 .. _lsp-clients-fortls-executable: https://emacs-lsp.github.io/lsp-mode/page/lsp-fortran/#lsp-clients-fortls-executable
 
+
+`Eglot <https://github.com/joaotavora/eglot>`__
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install the `eglot`_ package which supports fortls out of the box.
+This can be done in emacs version > 26.1 via ``M-x package-install RET eglot RET``.
+
+.. _eglot: https://github.com/joaotavora/eglot
```
Comment (gnikit) [inline docs/editor_integration.rst:179]: Can you specify how you pass settings to `fortls` via `eglot`

Would this work?

```
(add-to-list 'eglot-server-programs
             `(fortran-mode . ("fortls" "--notify_init" "--nthreads=2"
                              )))
```

Comment (ssnnoo) [reply docs/editor_integration.rst:179]: yes, I will adapt the pull request (hopefully) later today. Sorry for the delay.

**Review by gnikit — APPROVED** [review]
Great work @ssnnoo

**Pull request merged**